File: src/main/kotlin/org/psota/invoices/entity/Product.java
package org.psota.invoices.entity;

import jakarta.persistence.Entity;
import lombok.AccessLevel;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;
import lombok.experimental.SuperBuilder;

@ToString(callSuper = true)
@EqualsAndHashCode(onlyExplicitlyIncluded = true, callSuper = false)
@SuperBuilder(toBuilder = true)
@NoArgsConstructor(access = AccessLevel.PROTECTED) // for JPA
@Getter
@Entity
public class Product extends BaseEntity {

  private String name;
  private String description;
  private Unit unit;
  private Double unitPrice;

}

File: invoices-be/src/main/kotlin/org/psota/invoices/entity/Product.kt
package org.psota.invoices.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated

@Entity
class Product(
        @Column(nullable = false)
        val name: String,
        @Column(nullable = true)
        val description: String?,
        @Enumerated(EnumType.STRING)
        val unit: Unit,
        @Column(nullable = false)
        val unitPrice: Double
) : BaseEntity()
